### ISSUE TITLE
Video orientation on iPad is incorrect when starting video call in landscape mode

### ISSUE THREAD

**Issue by Manny-ModMed** [aws/amazon-chime-sdk-ios#88]
**Describe the bug**
When starting a video call on a iPad device and the device is in landscape mode, the video render of the user is not in the correct orientation. It is shown in portrait mode with the video stream being side ways. Now if the user rotates the device, the video orientation automatically corrects itself.

**To Reproduce**
1. Position an iPad in landscape mode.
2. Start a video call.
3. In the video renderer view of the local camera, notice how the orientation is not correct.

**Expected behavior**
The video orientation should match the device orientation when starting a video call.

**Logs**
N/A

**Screenshots**
![IMG_0004](https://user-images.githubusercontent.com/41126742/89858905-bc94ad80-db6d-11ea-9fb7-242ec97de769.PNG)

**Test environment Info (please complete the following information):**
 - Device: iPad 7th generation
 - OS: iOS 13.3.1
 - Version AmazonChimeSDK: 0.8.1
 - Version AmazonChimeSDKMedia: 0.6.0

**Additional context**
N/A

**Comment by hokyungh** [comment]
Thank you for reaching out. This seems very interesting bug. If you open camera (actual ipad camera app) and rerun the application, do you see orientation fixed?

**Comment by dchen-aws** [comment]
@Manny-ModMed I could not seem to be able to repro with my iPad with different combination. is this happening consistently for you? if so, can you share the repro steps?

**Comment by Manny-ModMed** [comment]
@hokyungh, @dchen-aws 

If I tried opening the default camera app and then re-running the application, the orientation is not fixed. Now if I rotate the device to portrait and then back to landscape, the orientation automatically updates. If I create a new video call within the same app instance/run, the orientation issue doesn't appear anymore. I believe this happens the first time the host application is running and a video call is created for the first time. I've attached a video showing the behavior that we are seeing in our POC:
[RPReplay_Final1597766032.MP4.zip](https://github.com/aws/amazon-chime-sdk-ios/files/5091445/RPReplay_Final1597766032.MP4.zip)

**Comment by linsang21** [comment]
Hey, by``` the first time the host application is running ``` you mean every time your app gets launched? 
So basically whenever you launch the app, and start your local video, the orientation is wrong. Is it only wrong in landscape mode, or also in portrait mode?
When you launch the default camera app before launching your app, is the orientation correct in the default camera app?
 
Also I'm interested that is there any configuration/interaction from your app to camera?

**Comment by JozefLipovsky** [comment]
Hi, I am able to reproduce this in my app and while testing the latest version of `AmazonChimeSDKDemo` as well. Every time when I start video after initial app launch while iPad is in landscape mode DefaultVideoRenderView image orientation is incorrect. When I rotate the iPad image orientation correct itself and when I leave the call and join again video image orientation remains correct. But after force kill and app restart the issue is still there. Launching default camera before launching `AmazonChimeSDKDemo` doesn't help. 

Everything is working fine after portrait mode launch.

![Xnip2020-10-02_17-49-26](https://user-images.githubusercontent.com/1158136/94979464-a8b75900-04d7-11eb-81bf-0dbd93f63785.jpg)

**Comment by JozefLipovsky** [comment]
Hi, any updates on this, please? I can still reproduce the issue in demo app, even after the latest release v0.16.0 update.

**Comment by Manny-ModMed** [comment]
Hello, any updates on this? We've recently used v0.16.1 and this is still reproducible in the demo application.

**Comment by JozefLipovsky** [comment]
As mentioned above relying on `UIDevice.current.orientation` in this use case is unreliable as it returns `.unknown` first time when the app is launched in the landscape mode (when you are using [apple smart folio case](https://www.apple.com/ca/shop/product/MXNK2LL/A/smart-keyboard-folio-for-ipad-pro-11-inch-3rd-generation-and-ipad-air-4th-generation-us-english?fnode=b003f9f987bc39356062be6939ed5aa66527f765ff9be90a74f91dd3c629ce35297cd1bd0c6dc56dc6939cc1a57a4b7a97f73456f89878f27a6a943ec9d8b552ab25ae67f756e812725008a8f91071164a928a4e6d8cac139b22b3c6fc18ea3ac68436d490a060c3b22400f04861347d) etc.)

In our app we also have a preview where our users can check their camera offline before we start a chime session and I had to fix exactly the same issue:

```
final class CameraPreviewView: UIView {
    override class var layerClass: AnyClass { AVCaptureVideoPreviewLayer.self }

    private var previewLayer: AVCaptureVideoPreviewLayer {
        guard let layer = layer as? AVCaptureVideoPreviewLayer else {
            fatalError("Expected `AVCaptureVideoPreviewLayer` type for layer.")
        }
        return layer
    }

    private weak var captureSession: AVCaptureSession?
    private var deviceOrientationObserver: NSObjectProtocol?

    init(captureSession: AVCaptureSession) {
        super.init(frame: .zero)
        self.captureSession = captureSession
        previewLayer.session = self.captureSession
        previewLayer.contentsGravity = .resizeAspectFill
        previewLayer.videoGravity = .resizeAspectFill
        previewLayer.connection?.automaticallyAdjustsVideoMirroring = false

        registerDeviceOrientationObserver()
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func registerDeviceOrientationObserver() {
         let notificationName = UIDevice.orientationDidChangeNotification
         deviceOrientationObserver = NotificationCenter.default.addObserver(forName: notificationName,
                                                                            object: nil,
                                                                            queue: .main) { [weak self] _ in
             guard self?.captureSession?.isRunning ?? false else {
                 return
             }

             self?.updateVideoOrientation()
         }
     }

    func removeDeviceOrientationObserver() {
         guard let deviceOrientationObserver = deviceOrientationObserver else { return }
         NotificationCenter.default.removeObserver(deviceOrientationObserver)
     }

    func updateVideoOrientation() {
        let deviceOrientation = self.window?.windowScene?.interfaceOrientation ?? .unknown
        var videoOrientation: AVCaptureVideoOrientation
        switch deviceOrientation {
            case .portrait:
                videoOrientation = .portrait
            case .landscapeLeft:
                videoOrientation = .landscapeLeft
            case .landscapeRight:
                videoOrientation = .landscapeRight
            case .portraitUpsideDown:
                videoOrientation = .portraitUpsideDown
            default:
                videoOrientation = .portrait
        }

        if previewLayer.connection?.videoOrientation != videoOrientation {
            previewLayer.connection?.videoOrientation = videoOrientation
        }
    }
}
```

I used `....windowScene?.interfaceOrientation` but `UIApplication.shared.statusBarOrientation` check will work as well since ChimeSDK still support older iOS versions.

I also spent some time trying to find a solution for this on our client side once we receive video frames via `VideoRenderView` protocol, but no luck. And I was looking literaly into the same thing as @Manny-ModMed yesterday 😆. I was able to make the demo app work as expected by changing `func updateOrientation()` to this:
```
    private func updateOrientation() {
        guard let connection = output.connection(with: AVMediaType.video) else {
            return
        }


        DispatchQueue.main.async {
            let deviceOrientation = UIApplication.shared.statusBarOrientation

            switch deviceOrientation {
                case .landscapeLeft:
                    connection.videoOrientation = .landscapeLeft
                case .landscapeRight:
                    connection.videoOrientation = .landscapeRight
                case .portrait:
                    connection.videoOrientation = .portrait
                case .portraitUpsideDown:
                    connection.videoOrientation = .portraitUpsideDown
                case .unknown:
                    connection.videoOrientation = .portrait
                @unknown default:
                    fatalError()
            }
        }


//        orientation = UIDevice.current.orientation

//        switch orientation {
//        case .portrait, .unknown:
//            connection.videoOrientation = .portrait
//        case .landscapeLeft:
//            connection.videoOrientation = .landscapeRight
//        case .portraitUpsideDown:
//            connection.videoOrientation = .portraitUpsideDown
//        case .landscapeRight:
//            connection.videoOrientation = .landscapeLeft
//        default:
//            // Do not update videoOrientation when the device is laying flat
//            return
//        }
    }
```

@hokyungh, @dchen-aws, @linsang21 can we please have someone on the amazon team to have look at this? We are planning to release our audio/video feature pretty soon, and this is one of the blockers for us for now. 

Thanks in advance!

**Comment by linsang21** [comment]
Hi there, thanks for the detailed updates and sharing the solution! I think @Manny-ModMed's solution makes sense. I'm wondering if you'd be interested in contributing your solution to the SDK, by forking the repo and send the PR from it? Of course our team take care of it if prefer otherwise.

**Comment by JozefLipovsky** [comment]
@linsang21 can someone please have a look #297

**Comment by Bengua27** [comment]
Hello, 
For the time being, I have added this before to activate my local video.
Not sexy at all, but it does the job
`
switch UIDevice.current.orientation {
        case .unknown, .faceUp, .faceDown:
            if let value = UIApplication.shared.appWindow()?.windowScene?.interfaceOrientation.rawValue {
                UIDevice.current.setValue(value, forKey: "orientation")
                UINavigationController.attemptRotationToDeviceOrientation()
            }
default:
            break
}
`

Hope it might help

**Comment by hokyungh** [comment]
This should have fixed in 0.16.3. If you still have still, feel free to reopen the issue. Thanks!
